src/log/state.ts

```diff
@@ -1,47 +1,43 @@
-import {LogLevels} from './levels';
-import {LogGroup} from './group';
+import {StrongBoolean, StrongMap, makeBoolean, makeInt, StrongInt} from '@toreda/strong-types';
 import {LogOptions} from '../log/options';
-import {StrongMap, StrongBoolean, makeBoolean} from '@toreda/strong-types';
+import {LogGroup} from './group';
+import {LogLevels} from './levels';
 
 /**
  * Holds internal state data, settings, and log groups for a
  * single log instance.
  */
+
 export class LogState extends StrongMap {
-	public globalLogLevel: number;
-	public readonly groups: Record<'all' | 'global' | string, LogGroup>;
-	public readonly groupKeys: string[];
-	public readonly groupsDefaultEnabled: StrongBoolean;
 	public readonly consoleEnabled: StrongBoolean;
+	public readonly globalLogLevel: StrongInt;
+	public readonly groupKeys: string[];
+	public readonly groups: Record<'all' | 'global' | string, LogGroup>;
+	public readonly groupsDisableOnStart: StrongBoolean;
 
 	constructor(options?: LogOptions) {
 		super();
-		this.globalLogLevel =
-			options && typeof options.globalLogLevel === 'number'
-				? options.globalLogLevel
-				: LogLevels.ALL & ~LogLevels.DEBUG & LogLevels.TRACE;
+
 		const defaultGroups = this.createDefaultGroups();
+
+		this.groups = defaultGroups.map;
+		this.groupKeys = defaultGroups.keys;
+
 		// Check whether groups should start enabled or disabled.
 		// Disabled groups do not process logs, even if the group log level
 		// or global log level would otherwise allow it.
-		const groupsEnabled =
-			options && typeof options.groupsDisableOnStart === 'boolean'
-				? !options.groupsDisableOnStart
-				: true;
-		this.groupsDefaultEnabled = makeBoolean(groupsEnabled, true);
-		this.groups = defaultGroups.map;
-		this.groupKeys = defaultGroups.keys;
+		this.groupsDisableOnStart = makeBoolean(true);
+
 		// Whether console output is enabled by default. If disabled,
 		// the built-in console transport can be activated at any time.
-		const enableConsole =
-			options && typeof options.consoleLogEnabled === 'boolean' ? options.consoleLogEnabled : false;
-		this.consoleEnabled = makeBoolean(enableConsole, false);
+		this.consoleEnabled = makeBoolean(false);
+
 		// Starting Global log level
-		if (options && typeof options.globalLogLevel === 'number') {
-			this.globalLogLevel = options.globalLogLevel;
-		}
+		this.globalLogLevel = makeInt(LogLevels.ALL & ~LogLevels.DEBUG & LogLevels.TRACE);
 
-		this.parse(options);
+		if (options != null) {
```

Comment:
Checking whether `options` is `null` should be unnecessary with the latest `@toreda/strong-types` package. `parse` should do nothing if the provided data is `undefined` or `null`.